src/hotspot/share/gc/serial/serialBlockOffsetTable.cpp

```diff
@@ -85,384 +81,109 @@ void BlockOffsetSharedArray::resize(size_t new_word_size) {
-  //
-  //    offset
-  //    card             2nd                       3rd
-  //     | +- 1st        |                         |
-  //     v v             v                         v
-  //    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+     +-+-+-+-+-+-+-+-+-+-+-
-  //    |x|0|0|0|0|0|0|0|1|1|1|1|1|1| ... |1|1|1|1|2|2|2|2|2|2| ...
-  //    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+     +-+-+-+-+-+-+-+-+-+-+-
-  //    11              19                        75
-  //      12
-  //
-  //    offset card is the card that points to the start of an object
-  //      x - offset value of offset card
-  //    1st - start of first logarithmic region
-  //      0 corresponds to logarithmic value N_words + 0 and 2**(3 * 0) = 1
-  //    2nd - start of second logarithmic region
-  //      1 corresponds to logarithmic value N_words + 1 and 2**(3 * 1) = 8
-  //    3rd - start of third logarithmic region
-  //      2 corresponds to logarithmic value N_words + 2 and 2**(3 * 2) = 64
-  //
-  //    integer below the block offset entry is an example of
-  //    the index of the entry
-  //
-  //    Given an address,
-  //      Find the index for the address
-  //      Find the block offset table entry
-  //      Convert the entry to a back slide
-  //        (e.g., with today's, offset = 0x81 =>
-  //          back slip = 2**(3*(0x81 - N_words)) = 2**3) = 8
-  //      Move back N (e.g., 8) entries and repeat with the
-  //        value of the new entry
-  //
-  size_t start_card = _array->index_for(start);
-  size_t end_card = _array->index_for(end-1);
-  assert(start ==_array->address_for_index(start_card), "Precondition");
-  assert(end ==_array->address_for_index(end_card)+BOTConstants::card_size_in_words(), "Precondition");
-  set_remainder_to_point_to_start_incl(start_card, end_card, reducing); // closed interval
-}
-
-
-// Unlike the normal convention in this code, the argument here denotes
-// a closed, inclusive interval: [start_card, end_card], cf set_remainder_to_point_to_start()
-// above.
-void
-BlockOffsetArray::set_remainder_to_point_to_start_incl(size_t start_card, size_t end_card, bool reducing) {
-
-  check_reducing_assertion(reducing);
-  if (start_card > end_card) {
-    return;
-  }
-  assert(start_card > _array->index_for(_bottom), "Cannot be first card");
-  assert(_array->offset_array(start_card-1) <= BOTConstants::card_size_in_words(),
-    "Offset card has an unexpected value");
-  size_t start_card_for_region = start_card;
-  u_char offset = max_jubyte;
-  for (uint i = 0; i < BOTConstants::N_powers; i++) {
-    // -1 so that the card with the actual offset is counted.  Another -1
-    // so that the reach ends in this region and not at the start
-    // of the next.
-    size_t reach = start_card - 1 + (BOTConstants::power_to_cards_back(i+1) - 1);
-    offset = BOTConstants::card_size_in_words() + i;
-    if (reach >= end_card) {
-      _array->set_offset_array(start_card_for_region, end_card, offset, reducing);
+// Write the backskip value for each region.
```

Comment:
```suggestion
// Write the backskip value for the given region.
```

Reply:
Note that region isn't defined very exactly. Here it is (apparently) the memory from `blk_start` to `blk_end`, but later it is also used as "area containing the same backskip value". Maybe there is some better wording to be found.

Reply:
"region" in this file always means "logarithmic region". If we refer to `[blk_start, blk_end]`, "block" should be used.

Reply:
I noticed just now. It would be nice to define that somewhere. :+1: